### DISCUSSION TITLE
Complex queries in Supabase

### DISCUSSION THREAD

**Discussion by theShreyans** [supabase/supabase#9403] [Questions]
So I'm in a situation where I'm trying to query somewhat complex relation tables. I'm not sure if this is possible in supabase yet but here's the current query I'm using.
```dart
await Supabase.instance.client
          .from('groups')
          .select(
              ' *,createdby:users(*), members(count)');
```
Here there is a column in groups called "createby" that is referenced to users table. And "members" table is connected to "groups" table by group(id).

What I'm trying to accomplish here is trying to get the details of a group + details of the person created the group + total number of people in the group.

So far I'm getting all that.

However the last piece that I'm trying to figure out is, lets say "Person A" decided to join a group. Now when that person refreshes the page I want to get "Person A" userId from "members" table, so I can tell "Person A" that they have already joined the group. So I was stuck trying to get this working. Does anyone know how to accomplish this?

Also, I'm using flutter.

**Comment by EdgarsLv** [comment]
How to perform this query with supabase client?

SELECT
  CASE 
    WHEN user_one = sender_id THEN user_two 
    ELSE user_one 
  END AS receiver_id
FROM conversations
WHERE user_one IS NOT NULL
  AND user_two IS NOT NULL
  AND updated_at > now() - interval '2 day'
  AND NOT updated_at > now() - interval '1 day;

And how to insert here text as code? example in comment above with colors and that can even copy it?

**Comment by steve-chavez** [comment]
> How to perform this query with supabase client

You can do that inside a SQL function and call it through RPC, see https://supabase.com/docs/reference/javascript/rpc.
